core/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java

```diff
@@ -196,34 +199,39 @@ protected boolean resolveIndex() {
 
         try {
             // add the response
-            final WriteResult<DeleteResponse> writeResult = TransportDeleteAction.executeDeleteRequestOnPrimary(deleteRequest, indexShard);
-            DeleteResponse deleteResponse = writeResult.getResponse();
-            location = locationToSync(location, writeResult.getLocation());
-            setResponse(item, new BulkItemResponse(item.id(), OP_TYPE_DELETE, deleteResponse));
+            WriteResult<DeleteResponse> writeResult = TransportDeleteAction.executeDeleteRequestOnPrimary(deleteRequest, indexShard);
+            if (writeResult.success()) {
+                DeleteResponse deleteResponse = writeResult.getResponse();
+                location = locationToSync(location, writeResult.getLocation());
+                setResponse(item, new BulkItemResponse(item.id(), OP_TYPE_DELETE, deleteResponse));
```

Comment:
I think this OP_TYPE_DELETE is not needed now that we have `Operation` (which was renamed to `Result`) on over DocWriteResponse. I think we can fold all this success/failure logic to a shard method and call it from both the indexing path and delete path? will remove a lot of boiler plate